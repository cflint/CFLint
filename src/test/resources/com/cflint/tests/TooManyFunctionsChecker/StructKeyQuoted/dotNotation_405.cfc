// not compliant
component{
	function x(){
		sampleStruct.yetAnotherKey = "yetAnotherValue";
		//repeat
		sampleStruct.yetAnotherKey = "yetAnotherValue";
		sampleStruct.foo["abc"] = "yetAnotherValue";
		sampleStruct.foo().xyzzy["abc"] = "yetAnotherValue";
		sampleStruct.foo().bar = "yetAnotherValue";
		sampleStruct.foo() = "yetAnotherValue";
	}
}