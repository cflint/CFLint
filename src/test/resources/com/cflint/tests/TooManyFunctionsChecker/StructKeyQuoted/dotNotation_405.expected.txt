{
  "version" : "",
  "timestamp" : 1526095032,
  "issues" : [ {
    "severity" : "WARNING",
    "id" : "STRUCT_ARRAY_NOTATION",
    "message" : "STRUCT_ARRAY_NOTATION",
    "category" : "CFLINT",
    "abbrev" : "SA",
    "locations" : [ {
      "file" : "src\\test\\resources\\com\\cflint\\tests\\TooManyFunctionsChecker\\StructKeyQuoted\\dotNotation_405.cfc",
      "fileName" : "dotNotation_405.cfc",
      "function" : "x",
      "offset" : 45,
      "column" : 3,
      "line" : 4,
      "message" : "Unquoted struct key yetAnotherKey is not case-sensitive.  Using array notation is recommended.",
      "variable" : "yetAnotherKey",
      "expression" : "sampleStruct.yetAnotherKey"
    } ]
  }, {
    "severity" : "WARNING",
    "id" : "STRUCT_ARRAY_NOTATION",
    "message" : "STRUCT_ARRAY_NOTATION",
    "category" : "CFLINT",
    "abbrev" : "SA",
    "locations" : [ {
      "file" : "src\\test\\resources\\com\\cflint\\tests\\TooManyFunctionsChecker\\StructKeyQuoted\\dotNotation_405.cfc",
      "fileName" : "dotNotation_405.cfc",
      "function" : "x",
      "offset" : 106,
      "column" : 3,
      "line" : 6,
      "message" : "Unquoted struct key yetAnotherKey is not case-sensitive.  Using array notation is recommended.",
      "variable" : "yetAnotherKey",
      "expression" : "sampleStruct.yetAnotherKey"
    } ]
  }, {
    "severity" : "WARNING",
    "id" : "STRUCT_ARRAY_NOTATION",
    "message" : "STRUCT_ARRAY_NOTATION",
    "category" : "CFLINT",
    "abbrev" : "SA",
    "locations" : [ {
      "file" : "src\\test\\resources\\com\\cflint\\tests\\TooManyFunctionsChecker\\StructKeyQuoted\\dotNotation_405.cfc",
      "fileName" : "dotNotation_405.cfc",
      "function" : "x",
      "offset" : 258,
      "column" : 3,
      "line" : 9,
      "message" : "Unquoted struct key bar is not case-sensitive.  Using array notation is recommended.",
      "variable" : "bar",
      "expression" : "sampleStruct.foo().bar"
    } ]
  } ],
  "counts" : {
    "totalFiles" : 0,
    "totalLines" : 0,
    "countByCode" : [ {
      "code" : "STRUCT_ARRAY_NOTATION",
      "count" : 3
    } ],
    "countBySeverity" : [ {
      "severity" : "WARNING",
      "count" : 3
    } ]
  }
}